scipy/optimize/minpack.py

```diff
@@ -449,23 +448,37 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         return (retval[0], info)
 
 
-def _wrap_func(func, xdata, ydata, weights):
-    if weights is None:
+def _wrap_func(func, xdata, ydata, T):
```

Comment:
@dlax I fixed this with more explicit documentation and change of name from T to transform in 7962556